File: examples/weather/testing/weather/plugins/src/weather/common/sign.go
package common

import (
	"bytes"
	"crypto/md5"
	"encoding/hex"
	"strconv"
	"strings"
)

func Sign(params ...interface{}) interface{} {
	var bs [][]byte
	if len(params) < 5 {
		return ""
	}
	i := params[0].(int64)
	ts := strconv.FormatInt(i, 10)
	i = params[1].(int64)
	nonce := strconv.FormatInt(i, 10)
	x := params[2].(int)
	id := strconv.Itoa(x)
	secret := params[3].(string)
	body := params[4].([]byte)
	s := []string{ts, nonce, id, secret}
	k := strings.Join(s, "")
	bs = append(bs, []byte(k), body)
	h := bytes.Join(bs, []byte(""))
	b := md5.Sum(h)
	return hex.EncodeToString(b[:])
}

File: testing/demo/plugins/src/pydemo/common/sign.py
import hashlib


def sign(ts, nonce, appid, secret: str, data: bytes):
    s = [ts.encode("utf-8"), nonce.encode("utf-8"), appid.encode("utf-8"), secret.encode("utf-8"), data]
    data = b"".join(s)
    k = hashlib.md5(data).hexdigest()
    print("sign:", k)
    assert 1==2, k
    return k
